### PR TITLE
Fix/dual legs transcribe race condition

### PR DESCRIPTION
**Pull request by xquanluu**

### CODE REVIEW

File: lib/tasks/transcribe.js [hunk 1]
```diff
@@ -263,7 +264,8 @@ class TaskTranscribe extends SttTask {
           throw new Error(`Invalid vendor ${this.vendor}`);
         }
     }
-
+    // save dedicated bugname for each endpoint
+    ep.transcribe_bugname = `${bugname}_${Date.now()}`;
```
Comment (davehorton) [inline lib/tasks/transcribe.js:268]: can we use a simple integer counter rather than Date.now() ?  It might be easier to debug if necessary

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/tasks/transcribe.js (hunk 1 above)
Comment (xquanluu) [reply lib/tasks/transcribe.js]: fixed

Comment (xquanluu) [reply lib/tasks/transcribe.js:269]: No We can't, due to issue that new transcribe verb can have the same name again to the old transcribe verb is being killed

File: lib/tasks/transcribe.js [hunk 2]
```diff
@@ -265,7 +266,8 @@ class TaskTranscribe extends SttTask {
         }
     }
     // save dedicated bugname for each endpoint
-    ep.transcribe_bugname = `${bugname}_${Date.now()}`;
+    ep.transcribe_bugname = `${bugname}_${channel}_${this.bugname_counter}`;
```
Comment (davehorton) [inline lib/tasks/transcribe.js]: maybe just 
```
ep.transcribe_bugname = `${bugname}_${channel}_${this.bugname_counter++}`;
```

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/tasks/transcribe.js (hunk 1 above)
Comment (davehorton) [reply lib/tasks/transcribe.js:269]: I'm confused though...seems like did make this change below?

File: lib/tasks/transcribe.js (hunk 2 above)
Comment (xquanluu) [reply lib/tasks/transcribe.js]: No, we can't use this, 

first verb will have transcribe_bugname_1_0
LCC to move the call to new transcribe verb in listen will have same bug name again transcribe_bugname_1_0

in the meanwhile the first transcribe is still being closed, the 2nd transcribe save private data to same bug name again

Still Date.now() is a must to make sure every single time we have different bugname

**Review by davehorton — APPROVED** [review]

**Pull request merged**